### PR TITLE
Provide support for native modules with ES6 imports

### LINKED ISSUE #3034: How to support builtin modules for ES6 import
Hi,

I have some builtin modules such as `gpio`, `i2c`, `spi`, etc and I want to know how to support ES6's `import from` like below:

```js
import {I2C} from 'i2c'
```

I had read https://jerryscript.net/ext-reference-module/ and tried `JERRYX_NATIVE_MODULE ` like below:

```c
static jerry_value_t i2c_module_on_resolve (void) {
  return ...;
}

JERRYX_NATIVE_MODULE (i2c, i2c_module_on_resolve)

void main() {
  ...
  i2c_register();
  ...
}
```

But `SyntaxError: Requested module not found. [<anonymous>:1:19]` error returned when try to use `import {I2C} from 'i2c'`.

Actually I already have my own module loader `require()`, so when user try to use `import` I want to call `require()` internally.

Any examples, advices or suggestions?

Thanks in advance.

### PR DESCRIPTION
**Pull request by dbatyai**
Resolves #3034.

JerryScript-DCO-1.0-Signed-off-by: Dániel Bátyai dbatyai@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/base/ecma-module.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ typedef enum
   ECMA_MODULE_STATE_PARSED = 2,     /**< module has been parsed */
   ECMA_MODULE_STATE_EVALUATING = 3, /**< module is currently being evaluated */
   ECMA_MODULE_STATE_EVALUATED = 4,  /**< module has been evaluated */
+  ECMA_MODULE_STATE_NATIVE = 5,     /**< module has been evaluated */
```
Comment (rerobika) [inline jerry-core/ecma/base/ecma-module.h]: Seems duplicated comment to me.

File: jerry-core/ecma/base/ecma-module.c [hunk 2]
```diff
@@ -114,16 +115,59 @@ ecma_module_find_or_create_module (ecma_string_t * const path_p) /**< module pat
...
+  return module_p;
+} /* ecma_module_create_module */
+
+/**
+ * Checks if we already have a module request in the module list.
+ *
+ * @return pointer to found or newly created module structure
+ */
+ecma_module_t *
+ecma_module_find_or_create_module (ecma_string_t * const path_p) /**< module path */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-module.c]: no space after `*`

File: jerry-core/include/jerryscript-compiler.h [hunk 3]
```diff
@@ -40,6 +40,7 @@ extern "C"
 #define JERRY_ATTR_NORETURN __attribute__((noreturn))
 #define JERRY_ATTR_PURE __attribute__((pure))
 #define JERRY_ATTR_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+#define JERRY_ATTR_WEAK __attribute__((weak))
```
Comment (zherczeg) [inline jerry-core/include/jerryscript-compiler.h]: What happens on a compiler other than gcc? The other attributes can be ignored, but this seems mandatory.

Comment (dbatyai) [reply jerry-core/include/jerryscript-compiler.h]: I reverted the weak attributes for now.

**Pull request merged**